PMSCore/Permission.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace PMSCore
+{
+    /// <summary>
+    /// Represents a user's access permission levels.
+    /// </summary>
+    public enum Permission
+    {
+        DEFAULT = 0,
```

Comment:
Maybe this should be more specific like

```c#
CanCreateProject
CanChangeProjectStatus
CanCrateTask
CanChangeTaskStatus
CanCreateUser
CanDeleteUser
CanGenerateReport
```

As some users can share Permissions, also permissions coudl be multiple per user like CanCreateProject and CanDeleteProject so maybe users could have HashSet<Permissions>